pyEpiabm/pyEpiabm/utility/state_transition_matrix.py

```diff
@@ -83,26 +109,33 @@ def update_probability(self, current_infection_status_row: InfectionStatus,
             the column where the probability will be updated
         new_probability : float
             Updated transition probability value
-        matrix : pd.Dataframe
-            Input state transition matrix that will have
-            one of its probabilities changed
 
         """
         try:
             if (current_infection_status_row not in InfectionStatus) or \
                     (next_infection_status_column not in InfectionStatus):
                 raise ValueError('Row and column inputs must be contained in' +
-                                 'the InfectionStatus enum')
+                                 ' the InfectionStatus enum')
         except TypeError:
             raise ValueError('Row and column inputs must be contained in' +
-                             'the InfectionStatus enum')
+                             ' the InfectionStatus enum')
 
         if (new_probability < 0) or (new_probability > 1):
             raise ValueError('New probability must be a valid probability' +
-                             'larger than or equal to 0 and less than' +
-                             'or equal to 1')
+                             ' larger than or equal to 0 and less than' +
+                             ' or equal to 1')
 
         # Extract row and column names from enum and retrieve trasition matrix
         row = current_infection_status_row.name
         column = next_infection_status_column.name
-        matrix.loc[row, column] = new_probability
+        self.matrix.loc[row, column] = new_probability
+
+    def remove_age_dependence(self):
```

Comment:
Well spotted- the age 40 is indeed a placeholder for age, when this is not being used in the simulation (similar to the placeholder of 0 we used before). However, if the the `use_ages` flag is `false` then this age is not used - instead we do indeed use a weighted average value across the population.

I personally don't think this needs significant justification - the point of the weighted average is actually to avoid the issues you had previously where the state transition matrix coefficients were not representative of the entire population - similarly using values for 40 -year olds would give a distorted matrix, underestimating the probabilities of higher risk infection states. This is all documented in the wiki of course, and the global nature of the `use_ages` flag means any method can verify which set of values are being used, but please do let me know if you don't find the explanation there clear!